File: NFLPool.Model/Team.cs
namespace NFLPool.Model
{
    public abstract class Team
    {
        public Team()
        {
            PoolName = string.Empty;
            Logo = string.Empty;
        }
        public string Name
        {
            get
            {
                return GetType().Name;
            }
            set
            {
                Name = value;
            }
        } 
        public string Logo { get; set; }
        public string PoolName { get; set; }
    }
}

File: java/src/main/java/com/effisfot/nflpool/model/Team.java
package com.effisfot.nflpool.model;

public abstract class Team {

    private String logo;
    private String poolName;

    public String getName() {
        return this.getClass().getSimpleName();
    }

    public String getLogo() {
        return logo;
    }

    public String getPoolName() {
        return poolName;
    }

    protected void setLogo(String logo) {
        this.logo = logo;
    }

    protected void setPoolName(String poolName) {
        this.poolName = poolName;
    }

}
